docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +275,115 @@ Note that:
+   the `c8y_configuration_plugin` downloads the new configuration content in `/tmp`,
+   and, on success, moves this content under the thin-edge file transfer HTTP root.
+   * The file is moved to `$TEDGE_HTTP_ROOT/config_snapshot/$CHILD_DEVICE_ID/$TYPE`
+     making this file available under `http://$TEDGE_HTTP/tedge/config_snapshot/$CHILD_DEVICE_ID/$TYPE`.
+   * Note that the `$TYPE` is by default the `$PATH` if not provided by the configuration.  
+1. Once the updated configuration is available over the local HTTP file transfer service,
+   the `c8y_configuration_plugin` notify the child device by publishing an MQTT message.
+   * The topic is `tedge/$CHILD_DEVICE_ID/commands/req/config_update`
+   * The payload is a JSON record with 3 fields
+     * `"url": "http://$TEDGE_HTTP/tedge/config_snapshot/$CHILD_DEVICE_ID/$PATH"`
+     * `"path": "$PATH"`
+     * `"type": "$TYPE"`
+1. On reception of a configuration update on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_update`,
+   The child-device specific software for configuration management:
+   1. Downloads the content from the `url` specified by the notification message.
+   1. Uses the `path` and `type` information to apply the new configuration content.
+      Note that these pieces of information are provided by the child-device specific software itself,
+      and make sense only in the specific context of the device operating system and software.
+1. While the configuration update is applied,
+   the child-device specific software for configuration management,
+   notifies thin-edge over MQTT about the progress of this operation.
+   1. These messages are published on the topic
+      `tedge/$CHILD_DEVICE_ID/commands/res/config_update`/<config file path>`
+   1. There is three different payloads to notify the configuration operation started,
+      finally succeed or failed for some reason.
+   1. All these payloads are JSON records with 3 required fields:
+      * `"status": "$STATUS"` where the status is either "EXECUTING", "SUCCESS" or "ERROR"
+      * `"path": "$PATH"`
+      * `"type": "$TYPE"`
+      * `"error": "$ERROR_MSG"` telling the cause of the error if any.
+   1. The child-device configuration software must send at least a success or an error message.
+      It should send an executing message.
+1. On reception an operation status message,
+   the `c8y_configuration_plugin` notifies the cloud accordingly.
+   1. When a success or error message is finally received,
+      then the configuration plugin cleans up all the temporary resources,
+      notably removing the file under `$TEDGE_HTTP_ROOT/config_snapshot/$CHILD_DEVICE_ID/$TYPE`.
+   1. If no responses are received from the child devices after 60s,
+      then a timeout error is sent to the cloud, and the resources cleaned.
+   1. If a notification message is received for a configuration file that is not under update
+      this message is ignored.
```

Comment:
```suggestion
   1. If a notification message is received for a configuration file that is not under update topic
      this message is ignored.
```

Reply:
I have to improve the wording as you suggestion highlights a confusion.

The point is about "what if the `c8y_configuration_plugin` receives a notification, say `{ "path":"foo", "status": "success"}`,
while the plugin were *not* waiting a notification for this path?". This might arrive after a timeout.